presto-main/src/test/java/com/facebook/presto/operator/scalar/FunctionAssertions.java

```diff
@@ -751,7 +752,13 @@ public static Expression createExpression(Session session, String expression, Me
     {
         Expression parsedExpression = SQL_PARSER.createExpression(expression, createParsingOptions(session));
 
-        parsedExpression = rewriteIdentifiersToSymbolReferences(parsedExpression);
+        if (metadata.getTypeManager().getTypes().stream().noneMatch(EnumType.class::isInstance)) {
```

Comment:
Let's not do this. You can add a flag to enable / disable this at the very least. This logic doesn't belong here. Please keep interfaces clean.

The way I prefer would be to change the implementation of `rewriteIdentifiersToSymbolReferences` and add a `rewriteDereference` to check `if (base instanceof Identifier) return node;`

Reply:
this direction ended up being pretty complicated to get tests working, so instead I'm going for a test file that uses a full-fledged DistributedQueryRunner to execute enum tests